ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeSignatureValidator.java

```diff
@@ -151,6 +158,21 @@ private InternalValidationResult validateWithState(
       return REJECT;
     }
 
+    // [IGNORE] There has been no other valid sync committee signature for the declared slot for the
+    // validator referenced by sync_committee_signature.validator_index.
+    final UniquenessKey uniquenessKey =
+        maybeUniquenessKey.orElseGet(
+            () -> {
+              final int subnetId =
+                  assignedSubcommittees.getAssignedSubcommittees().stream()
+                      .min(Integer::compare)
+                      .orElseThrow();
+              return getUniquenessKey(signature, subnetId);
```

Comment:
This makes me a bit nervous.  Given we'll be publishing the signature to all subnets, I wonder if we should mark it as seen on all applicable subnets as well.  It would be quite reasonable to always publish locally produced signatures though - libp2p will still filter out duplicates at the lower level if they're identical which they should be unless something is going very wrong.

Alternatively we could `IGNORE` if all applicable subnets are already marked as seen which is probably the most accurate way to handle this case.

Reply:
@mbaxter Wanted to get this merged so I could launch the next Altair devnet so have added https://github.com/ConsenSys/teku/pull/4088/commits/6fa52ad6f22e1cb53408f0e9b8fa2fdd352fb71b to make this check all applicable subnets for local validator keys.  It came out pretty neatly I think so probably worth getting the detail right.  Happy to revert or rework it if you aren't happy with it when you get back though.

Reply:
Looks good - thanks!